airflow/cli/commands/task_command.py

```diff
@@ -149,23 +150,27 @@ def _get_dag_run(
             execution_date=dag_run_execution_date,
             run_id=_generate_temporary_run_id(),
             data_interval=dag.timetable.infer_manual_data_interval(run_after=dag_run_execution_date),
-            session=session,
+            **session_kwarg,
         )
         return dag_run, True
     raise ValueError(f"unknown create_if_necessary value: {create_if_necessary!r}")
 
 
-@provide_session
 def _get_ti(
     task: Operator,
     map_index: int,
     *,
     exec_date_or_run_id: str | None = None,
     pool: str | None = None,
     create_if_necessary: CreateIfNecessary = False,
-    session: Session = NEW_SESSION,
+    session: Session | None = None,
 ) -> tuple[TaskInstance | TaskInstancePydantic, bool]:
     """Get the task instance through DagRun.run_id, if that fails, get the TI the old way."""
+    # this is required by AIP-44; if there's a session here, let's use it
+    # if it's None, don't pass it (because @provide_session does not check
+    # whether the provided session is None or not
+    session_kwarg = {"session": session} if session else {}
```

Comment:
Not in love with this repeated pattern.

Reply:
Me neither.  Have to figure something out. One option is make provide session check if null

Reply:
ok PTAL i have updated this to just non deal with sessions in _get_ti and let the called methods create and close their own sessions.